frontends/web/src/routes/exchange/components/buysell.tsx

```diff
@@ -0,0 +1,155 @@
+import { useNavigate } from 'react-router-dom';
+import { useLoad } from '@/hooks/api';
+import * as exchangesAPI from '@/api/exchanges';
+import { useContext, useEffect, useState } from 'react';
+import { ExchangeSelectionRadio } from './exchangeselectionradio';
+import { Button } from '@/components/forms/button';
+import { useTranslation } from 'react-i18next';
+import { getVersion } from '@/api/bitbox02';
+import style from '../exchange.module.css';
+import { AppContext } from '@/contexts/AppContext';
+import { TInfoContentProps } from './infocontent';
+
+type TProps = {
+  accountCode: string;
+  deviceIDs: string[];
+  selectedRegion: string;
+  onSelectExchange: (exchange: string) => void;
+  selectedExchange: string;
+  goToExchange: () => void;
+  showBackButton: boolean;
+  action: exchangesAPI.TExchangeAction
+  setInfo: (into: TInfoContentProps) => void;
+}
+
+export const BuySell = ({
+  accountCode,
+  deviceIDs,
+  selectedRegion,
+  onSelectExchange,
+  selectedExchange,
+  goToExchange,
+  showBackButton,
+  action,
+  setInfo,
+}: TProps) => {
+  const { t } = useTranslation();
+  const { setFirmwareUpdateDialogOpen } = useContext(AppContext);
+
+  const exchangeDeals = useLoad(() => exchangesAPI.getExchangeDeals(action, accountCode, selectedRegion), [action, selectedRegion]);
+  const [exchangeDealsError, setExchangeDealsError] = useState('');
+  const [updateFirmwareToSell, setUpdateFirmwareToSell] = useState(false);
+  // FIXME: we should make a BE `can-sell` endpoint
+  const versionInfo = useLoad(() => getVersion(deviceIDs[0]), [deviceIDs[0]]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
```

Comment:
Can we add short comments on what each of them does here?:)

Tbh I'm not the biggest fan of using many `useEffect`s in one file (as it can be kinda hard to read) but sometimes it's unavoidable.

We don't hv to do it here, but we can turn these into separate hooks e.g  `useCheckExchangeDealsError()`, `useCheckFirmwareUpdate()` and so on (with those comments) for better readability.

Reply:
I removed the whole `exchangeDealsError ` state and am now storing the whole `exchangeDealsResponse` as a state containing both legitimate results and errors. This allowed to remove one of the `useEffects`.

I didn't touch the firmware part as I expect this to be changed to a `useLoad()` that stores a bool for 'supportsSilentPayments' or 'canSell' or smth similar.